From 0fe9b48bad7094b62ebc368af5025d625c5caf6c Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:02:41 -0400
Subject: [usb] ohci: fix and explain sparse errors

Message-id: <1410887463-20674-90-git-send-email-dzickus@redhat.com>
Patchwork-id: 95119
O-Subject: [RHEL7 PATCH 089/591] USB: OHCI: fix and explain sparse errors
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 7c71c6e6befd4b234e166f2f40d4430bacc779e6
Author: Alan Stern <stern@rowland.harvard.edu>
Date:   Fri Oct 18 11:16:07 2013 -0400

    USB: OHCI: fix and explain sparse errors

    This patch fixes an endian-related error in ohci-hcd (detected by
    sparse) and clarifies a comment explaining a peculiar locking
    arrangement that sparse warns about.

    Signed-off-by: Alan Stern <stern@rowland.harvard.edu>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/host/ohci-hub.c b/drivers/usb/host/ohci-hub.c
index 2347ab8..61705a7 100644
--- a/drivers/usb/host/ohci-hub.c
+++ b/drivers/usb/host/ohci-hub.c
@@ -212,10 +212,11 @@ __acquires(ohci->lock)
  /* Sometimes PCI D3 suspend trashes frame timings ... */
  periodic_reinit (ohci);
 
- /* the following code is executed with ohci->lock held and
-  * irqs disabled if and only if autostopped is true
+ /*
+  * The following code is executed with ohci->lock held and
+  * irqs disabled if and only if autostopped is true.  This
+  * will cause sparse to warn about a "context imbalance".
   */
-
 skip_resume:
  /* interrupts might have been disabled */
  ohci_writel (ohci, OHCI_INTR_INIT, &ohci->regs->intrenable);
@@ -531,7 +532,7 @@ ohci_hub_descriptor (
      temp |= 0x0010;
  else if (rh & RH_A_OCPM) /* per-port overcurrent reporting? */
      temp |= 0x0008;
- desc->wHubCharacteristics = (__force __u16)cpu_to_hc16(ohci, temp);
+ desc->wHubCharacteristics = cpu_to_le16(temp);
 
  /* ports removable, and usb 1.0 legacy PortPwrCtrlMask */
  rh = roothub_b (ohci);
-- 
1.7.1